Restartless

derinb opened the pull request:
Restartless version of Personas Plus

wagnerand commented:
Please deduplicate the locales. We generally don't want any duplicate resources.

wagnerand commented on extension/chrome.manifest:
```diff
@@ -95,4 +81,4 @@ locale   personas   tr          locale/tr/
 locale   personas   vi          locale/vi/
 locale   personas   zh-CN       locale/zh-CN/
-locale   personas   zh-TW       locale/zh-TW/
+locale   personas   zh-TW       locale/zh-TW/
```
Please add a newline at the end of the file.

wagnerand commented on extension/chrome.manifest:
```diff
@@ -1,11 +1,3 @@
-# Register a resource: protocol alias so we can refer to files in this package
-# with URLs of the form resource://personas/path/to/file,
-# f.e. resource://personas/modules/service.js.
-resource personas   ./
```
You are still using the `resource` protocol in the code.

derinb replied:
Because resource protocol is registered by bootstrap.js, I removed it from chrome.manifest not to create conflicts.

wagnerand replied:
Ok.

wagnerand commented on extension/defaults/preferences/prefs.js:
```diff
@@ -121,3 +121,7 @@ pref("PersonaService.logging.dump", "None");
 // Whether or not the toolbar button has been installed once (Firefox 4 support)
 pref("extensions.personas.toolbarButtonInstalled", false);
+
+pref("extensions.personas.tbadded",false);
```
This seems unused.

derinb replied:
Sure will remove it.

wagnerand commented on extension/defaults/preferences/prefs.js:
```diff
@@ -121,3 +121,7 @@ pref("PersonaService.logging.dump", "None");
+
+pref("extensions.personas.tbadded",false);
+pref("extensions.personas.tbinsert",true);
+pref("extensions.personas.tbplaced",false);
```
Why do you need both `tbinsert` and `tbplaced`?

Also, newline please.

derinb replied:
`tbinsert` controls if the toolbar button should be inserted into UI. Say, if user removes it from UI, the button will not be added to UI anymore by the help of `tbinsert`.

`tbplaced` controls the placement of the toolbar button. At first install, button is placed and preference is saved. So add-on will not try to place it again and again by the help of `tbplaced` preference value.

wagnerand replied:
I still think those can be merged into one pref.

wagnerand commented on extension/install.rdf:
```diff
@@ -6,7 +6,8 @@
     <em:description>Your Firefox, your style. Easy-to-change themes.</em:description>
-
+	<em:bootstrap>true</em:bootstrap>
+	<em:unpack>true</em:unpack>
```
I don't see a reason for this to be unpacked and it can cause some troubles with signing, so let's not do that.

derinb replied:
Sure, just thought maybe you might want to check the add-on in Profile folder after it is compiled. So left it as true. Will remove it.

wagnerand replied:
Also, fix the indentation, please.

wagnerand commented:
All the css `!important` changes seem wrong. There must be another way to achieve this.

wagnerand commented:
`extensions/content/personas.js` is still there.

wagnerand commented on extension/modules/personas.js:
```diff
@@ -0,0 +1,1480 @@
+
+Components.utils.import("resource://gre/modules/Services.jsm");
+try{var console = (Cu.import("resource://gre/modules/Console.jsm", {})).console;}
+catch(error){var console = (Cu.import("resource://gre/modules/"+"devtools"+"/Console.jsm", {})).console;}
```
Proper indentation and formatting, please.

wagnerand commented on extension/modules/personas.js:
```diff
@@ -0,0 +1,1480 @@
+// It's OK to import the service module into the global namespace because its
+// exported symbols all contain the word "persona" (f.e. PersonaService).
+Cu.import("resource://personas/modules/service.js");
+	
```
Trailing spaces, also in other places. Please check all your files.

wagnerand commented on extension/modules/personas.js:
```diff
@@ -0,0 +1,1480 @@
+        delete this._menu;
+        return this._menu = document.getElementById("personas-menu");
+    },
+    get_menu(document) {
```
Why did you duplicate all this methods instead of just re-using the ones already present?

derinb replied:
Because old `content/personas.js` used to be injected into every browser `window` by `overlay.xul`. It saved properties into the `browser.xul` `window` object. 

But current `modules/personas.js` works in `bootstrap.js`. Add-on first detects the calling `document` from `event` and then request the `menu` via `get_menu` method by the help of that passed `document`.

I just duplicated main algorithm with a similar name so it can be compared easily.

wagnerand replied:
Please cleanup this code. Again, comparing is not done inside the file but by diff-tools.

derinb replied:
Do you mean to remove those getter methods that are not used?

wagnerand replied:
I mean deduplicating the code.

On Mon, May 30, 2016 at 2:07 PM, derinb <notifications@github.com> wrote:

> In extension/modules/personas.js
> <https://github.com/mozilla/personas-plus/pull/24#discussion_r65063095>:
>
> > +    get _strings() {
> > +        delete this._strings;
> > +        return this._strings = new this.StringBundle("chrome://personas/locale/personas.properties");
> > +    },
> > +
> > +    get _brandStrings() {
> > +        delete this._brandStrings;
> > +        return this._brandStrings =
> > +            new this.StringBundle("chrome://branding/locale/brand.properties");
> > +    },
> > +
> > +    get _menu() {
> > +        delete this._menu;
> > +        return this._menu = document.getElementById("personas-menu");
> > +    },
> > +    get_menu(document) {
>
> Do you mean to remove those getter methods that are not used?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24/files/0e46a799888a86b66555e538330edc80a3fc6aac#r65063095>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKi-2QAl2jIxL_OozZADZML58fz8lks5qGtLzgaJpZM4Iout8>
> .
>

derinb replied:
But those are getter functions. Getter functions does not accept parameters.

I just leaved getter functions so that you can compare the difference.

wagnerand replied:
Again, please *stop* doing that. If you rewrite code, delete the old one.
Diff-tools are supposed to handle diffs, and they are actually better at it
than humans.

On Mon, May 30, 2016 at 2:15 PM, derinb <notifications@github.com> wrote:

> In extension/modules/personas.js
> <https://github.com/mozilla/personas-plus/pull/24#discussion_r65063839>:
>
> > +    get _strings() {
> > +        delete this._strings;
> > +        return this._strings = new this.StringBundle("chrome://personas/locale/personas.properties");
> > +    },
> > +
> > +    get _brandStrings() {
> > +        delete this._brandStrings;
> > +        return this._brandStrings =
> > +            new this.StringBundle("chrome://branding/locale/brand.properties");
> > +    },
> > +
> > +    get _menu() {
> > +        delete this._menu;
> > +        return this._menu = document.getElementById("personas-menu");
> > +    },
> > +    get_menu(document) {
>
> But those are getter functions. Getter functions does not accept
> parameters.
>
> I just leaved getter functions so that you can compare the difference.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24/files/0e46a799888a86b66555e538330edc80a3fc6aac#r65063839>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKmgyVXZpaiAujiyeU_fRsb6bRgWgks5qGtTzgaJpZM4Iout8>
> .
>

derinb replied:
OK. Removed getter functions.

wagnerand commented on extension/modules/personas.js:
```diff
@@ -0,0 +1,1480 @@
+
+    openURLInTab: function(url,event) {
+		var document=event.currentTarget.ownerDocument;
+		var window=document.defaultView;
```
Please use distinct variable names, instead of just overriding `document` and `window`.

derinb replied:
`extension/modules/personas.js` is loaded into `bootstrap.js` context. `persona.js` is not injected into windows. So there is no `window` or `document` conflict over here.

We detect the `window` and `document` by the help of `event` that take place.

wagnerand replied:
Oh, right. Thanks.

wagnerand commented on extension/modules/personas.js:
```diff
@@ -0,0 +1,1480 @@
+	THUNDERBIRD_ID: "{3550f703-e582-4d05-9a08-453d09bdfdc6}",
+	FIREFOX_ID: "{ec8030f7-c20a-464f-9b0e-13a3a9e97384}",
+	SEAMONKEY_ID:"{92650c4d-4b8e-4d2a-b7eb-24ecf4f6b63a}",	
+	appInfo: Components.classes["@mozilla.org/xre/app-info;1"].getService(Components.interfaces.nsIXULAppInfo),		
```
Nit: Please deduplicate.

derinb replied:
Which duplicates do you mean?

wagnerand replied:
Those constants are defined in `extension/modules/service.jsextension/modules/service.js`.

wagnerand commented:
Please use four-space indentation instead of `tab`.

derinb commented:
Hello Andreas,

What do you mean by 

> Please deduplicate the locales. We generally don't want any duplicate resources.

derinb commented:
> All the css !important changes seem wrong. There must be another way to achieve this.

I seen many times Firefox's theme css overwrites add-on's  css rules. I just added `!important` to make sure Personas add-on related rules wins for all cases. 

It works even without `!important` for Windows. But I did not test all linux and old macs rules.

So to make sure it can work for all users and themes, I added `!important` as precaution. Anyway the rules are only Personas Plus related and does not leak into browser.css.

wagnerand commented on extension/modules/personas.js:
```diff
@@ -0,0 +1,1480 @@
+		else ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",true);
+	},
+	addToolsMenu:function(window){
+		var document=window.document;
```
I think we can drop this and use the toolbar button exclusively.

derinb replied:
Do you mean to remove Tools menu? I think some users will want to keep it.

wagnerand replied:
We should be using the toolbar button only.

On Mon, May 30, 2016 at 2:04 PM, derinb <notifications@github.com> wrote:

> In extension/modules/personas.js
> <https://github.com/mozilla/personas-plus/pull/24#discussion_r65062866>:
>
> > +			else{		
> > +				var navbar = (document.getElementById("nav-bar") || document.getElementById("mail-bar3") || document.getElementById("msgToolbar"));
> > +				navbar.insertItem(toolbarbutton.id, null, null, false);			
> > +			}
> > +			if(!document.getElementById(toolbarbutton.id)) (toolbar || navbar).insertItem(toolbarbutton.id, null, null, false);				
> > +		}
> > +	},
> > +	afterCustomization:function(event){
> > +		var window=event.currentTarget;
> > +		var document=window.document;
> > +		var ythdprefsinstance = Components.classes["@mozilla.org/preferences-service;1"].getService(Components.interfaces.nsIPrefBranch);
> > +		if(!document.getElementById("personas-toolbar-button") || document.getElementById("personas-toolbar-button").parentNode.id=="BrowserToolbarPalette") ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",false);
> > +		else ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",true);
> > +	},
> > +	addToolsMenu:function(window){
> > +		var document=window.document;
>
> Do you mean to remove Tools menu? I think some users will want to keep it.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24/files/0e46a799888a86b66555e538330edc80a3fc6aac#r65062866>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKhf4bNnJ9caJnyFz8Qg2yTY1GTCiks5qGtJogaJpZM4Iout8>
> .
>

derinb replied:
OK, but some users remove Toolbar button to preserve space and only use Tools menu. As of now, current version of Personas Plus on AMO allows that. When Tools menu removed, they may not be able to use the add-on if the toolbar button is removed by user.

What do you think?

wagnerand replied:
I get that, but don't you think the code would be much cleaner without it?
It this was a jetpack add-on or even a webextension (which it might become
eventually), we'd also have a toolbar button only.

On Mon, May 30, 2016 at 3:46 PM, derinb <notifications@github.com> wrote:

> In extension/modules/personas.js
> <https://github.com/mozilla/personas-plus/pull/24#discussion_r65074023>:
>
> > +			else{		
> > +				var navbar = (document.getElementById("nav-bar") || document.getElementById("mail-bar3") || document.getElementById("msgToolbar"));
> > +				navbar.insertItem(toolbarbutton.id, null, null, false);			
> > +			}
> > +			if(!document.getElementById(toolbarbutton.id)) (toolbar || navbar).insertItem(toolbarbutton.id, null, null, false);				
> > +		}
> > +	},
> > +	afterCustomization:function(event){
> > +		var window=event.currentTarget;
> > +		var document=window.document;
> > +		var ythdprefsinstance = Components.classes["@mozilla.org/preferences-service;1"].getService(Components.interfaces.nsIPrefBranch);
> > +		if(!document.getElementById("personas-toolbar-button") || document.getElementById("personas-toolbar-button").parentNode.id=="BrowserToolbarPalette") ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",false);
> > +		else ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",true);
> > +	},
> > +	addToolsMenu:function(window){
> > +		var document=window.document;
>
> OK, but some users remove Toolbar button to preserve space and only use
> Tools menu. As of now, current version of Personas Plus on AMO allows that.
> When Tools menu removed, they may not be able to use the add-on if the
> toolbar button is removed by user.
>
> What do you think?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24/files/0e46a799888a86b66555e538330edc80a3fc6aac#r65074023>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKqsiq39WDPYbCxhoBQETN00DAg63ks5qGuowgaJpZM4Iout8>
> .
>

derinb replied:
Yes I know but Firefox users are power users and with significant some of them use the Tools menu. Especially small screen laptop users have to remove Toolbar Button to preserve space and rely on Tools menu. So Tools menu is very important for them.

wagnerand replied:
Without having actual data to prove this, I'd argue that Personas-Plus
users are not necessarily power users. Since you've already written the
code, I guess we can leave it in for now, but sooner or later, this should
go away.

On Mon, May 30, 2016 at 4:03 PM, derinb <notifications@github.com> wrote:

> In extension/modules/personas.js
> <https://github.com/mozilla/personas-plus/pull/24#discussion_r65075858>:
>
> > +			else{		
> > +				var navbar = (document.getElementById("nav-bar") || document.getElementById("mail-bar3") || document.getElementById("msgToolbar"));
> > +				navbar.insertItem(toolbarbutton.id, null, null, false);			
> > +			}
> > +			if(!document.getElementById(toolbarbutton.id)) (toolbar || navbar).insertItem(toolbarbutton.id, null, null, false);				
> > +		}
> > +	},
> > +	afterCustomization:function(event){
> > +		var window=event.currentTarget;
> > +		var document=window.document;
> > +		var ythdprefsinstance = Components.classes["@mozilla.org/preferences-service;1"].getService(Components.interfaces.nsIPrefBranch);
> > +		if(!document.getElementById("personas-toolbar-button") || document.getElementById("personas-toolbar-button").parentNode.id=="BrowserToolbarPalette") ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",false);
> > +		else ythdprefsinstance.setBoolPref("extensions.personas.tbinsert",true);
> > +	},
> > +	addToolsMenu:function(window){
> > +		var document=window.document;
>
> Yes I know but Firefox users are power users and with significant some of
> them use the Tools menu. Especially small screen laptop users have to
> remove Toolbar Button to preserve space and rely on Tools menu. So Tools
> menu is very important for them.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24/files/0e46a799888a86b66555e538330edc80a3fc6aac#r65075858>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKs_iDWVoHTfS8OXUIOZby9Sr5f80ks5qGu4ZgaJpZM4Iout8>
> .
>

wagnerand commented:
All personas_bootstrap.properties` files are duplicates of the respective `personas.dtd`.

wagnerand commented:
Please use proper formatting and indenting in all of your code. Also, please make your code more readable, by not writing conditional blocks in just one line.

derinb commented:
> All personas_bootstrap.propertiesfiles are duplicates of the respective personas.dtd`.

Toobar menu and menuitem labels locales are contained in personas.dtd. Because we can not use `personas.dtd` in bootstrap add-ons, it had to be converted to `personas_bootstrap.properties`.

I know there are some entities that belong to customPersonasEditor.xul but I am also thinking to port customPersonasEditor.xul into browser so in future those entities will be used. 

Also there are 41 languages and it is very difficult to manually edit all entities. So it be great if it is accepted this way.

wagnerand commented:
The problem is that locales will become super hard to have be correct as we
now have to keep track of the same string in multiple places. We really
shouldn't be doing that.

On Mon, May 30, 2016 at 2:01 PM, derinb <notifications@github.com> wrote:

> All personas_bootstrap.propertiesfiles are duplicates of the respective
> personas.dtd`.
>
> Toobar menu and menuitem labels locales are contained in personas.dtd.
> Because we can not use personas.dtd in bootstrap add-ons, it had to be
> converted to personas_bootstrap.properties.
>
> I know there are some entities that belong to customPersonasEditor.xul but
> I am also thinking to port customPersonasEditor.xul into browser so in
> future those entities will be used.
>
> Also there are 41 languages and it is very difficult to manually edit all
> entities. So it be great if it is accepted this way.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24#issuecomment-222477611>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKrpNpEsKYaigFdgdwzS4x5ys5V8uks5qGtGDgaJpZM4Iout8>
> .
>

derinb commented:
> The problem is that locales will become super hard to have be correct as we now have to keep track of the same string in multiple places. We really shouldn't be doing that.

I definitely see your point. But I have to go and manually modify the personas.dtd files. During that if the files is saved with wrong encoding or if I introduce a typo, it will not be good.

Event the file structure is different among locales:
https://github.com/mozilla/personas-plus/blob/master/extension/locale/en-US/personas.dtd
https://github.com/mozilla/personas-plus/blob/master/extension/locale/de/personas.dtd

And in future, if the deleted entities be used in customPersonasEditor.xul file, we have to re-introduce the deleted entities which will be unnecessary.

I just wanted to make the transition with less modification and issues possible. Kept the personas.dtd as it is and created a new properties file.

If you still think it is necessary, I will go through the files. Also if there is bugzilla submission, it will be more secure the delete entities from web interface.

derinb commented:
> Please use proper formatting and indenting in all of your code. Also, please make your code more readable, by not writing conditional blocks in just one line.

Hello Andreas,

Converted tabs to spaces. Any left issues please?

wagnerand commented:
> If you still think it is necessary, I will go through the files.

Yes, we should only have one set of files, without any duplication.
Otherwise, it will be very bad for localizers.

Please note that I just merged a locale update (
https://github.com/mozilla/personas-plus/commit/88fc12956c2624b624fa8e5eadc2596f14032a7f).
I realize that might give you some extra work, but there was no other way
as the localized strings come from an external platform and they wouldn't
have applied anymore. Please rebase your patch.

> Also if there is bugzilla submission, it will be more secure the delete
entities from web interface.

I don't follow. Can you elaborate?

On Mon, May 30, 2016 at 2:57 PM, derinb <notifications@github.com> wrote:

> The problem is that locales will become super hard to have be correct as
> we now have to keep track of the same string in multiple places. We really
> shouldn't be doing that.
>
> I definitely see your point. But I have to go and manually modify the
> personas.dtd files. During that if the files is saved with wrong encoding
> or if I introduce a typo, it will not be good.
>
> Event the file structure is different among locales:
>
> https://github.com/mozilla/personas-plus/blob/master/extension/locale/en-US/personas.dtd
>
> https://github.com/mozilla/personas-plus/blob/master/extension/locale/de/personas.dtd
>
> And in future, if the deleted entities be used in customPersonasEditor.xul
> file, we have to re-introduce the deleted entities which will be
> unnecessary.
>
> I just wanted to make the transition with less modification and issues
> possible. Kept the personas.dtd as it is and created a new properties file.
>
> If you still think it is necessary, I will go through the files. Also if
> there is bugzilla submission, it will be more secure the delete entities
> from web interface.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24#issuecomment-222487283>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKiJDg6gII4gr8y68xH1uejT4oiUiks5qGt7EgaJpZM4Iout8>
> .
>

derinb commented:
Oh I meant the Babelzilla, sorry.. Where do localizations originate from?

wagnerand commented:
They come from Babelzilla.

derinb commented:
OK, now should I keep personas_bootstrap.properties file? Or append the required strings into personas.properties?

wagnerand commented:
Whatever works for you, without duplicating any of the strings.

On Mon, May 30, 2016 at 4:45 PM, derinb <notifications@github.com> wrote:

> OK, now should I keep personas_bootstrap.properties file? Or append the
> required strings into personas.properties?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24#issuecomment-222507443>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKuDGDehpXEyk265a3KjEbq_pN-oIks5qGvgjgaJpZM4Iout8>
> .
>

derinb commented:
Thanks. Do I need to go and remove those old strings from `personas.dtd`, too?

wagnerand commented:
I don't know what you mean by that. But I guess I already answered it when
I said "without duplicating any of the strings".

On Mon, May 30, 2016 at 4:57 PM, derinb <notifications@github.com> wrote:

> Thanks. Do I need to go and remove those old strings from personas.dtd,
> too?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24#issuecomment-222509568>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKlNLm27-iqhdkLxPBHD4E3i1qra0ks5qGvrngaJpZM4Iout8>
> .
>

derinb commented:
Andreas, I submitted locale changes but it did not allow me to push changes because of the conflict.

https://github.com/mozilla/personas-plus/pull/24/commits/6cff54b814cb37934fcfd6a1e98fdefb94a0f4d3

After Sync, it pushed but there were unnecessary commits included in this PR.

Any advice please?

wagnerand commented:
As I said, you need to rebase your PR onto master.

On Mon, May 30, 2016 at 7:24 PM, derinb <notifications@github.com> wrote:

> Andreas, I submitted locale changes but it did not allow me to push
> changes because of the conflict.
>
> 6cff54b
> <https://github.com/mozilla/personas-plus/pull/24/commits/6cff54b814cb37934fcfd6a1e98fdefb94a0f4d3>
>
> After Sync, it pushed but there were unnecessary commits included in this
> PR.
>
> Any advice please?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/personas-plus/pull/24#issuecomment-222530947>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA1TKgUJHX6t2QE9jf_5TxTj4L8iUw8wks5qGx1SgaJpZM4Iout8>
> .
>

derinb commented:
Yes I rebased, but it does not allow me to Push because of conflicts.

Then I Sync, but this time it adds uncessary commits in my PR.

Same vicious cycle.

derinb commented:
If you are OK with Locale Updates, I will close this PR and upload another.
https://github.com/mozilla/personas-plus/pull/24/commits/e7500cc79dce57e8962c1e4e3335cd580ec63ab4

Removed duplicate entries from DTD, and added missing chrome.manifest

wagnerand commented:
Please don't. This PR contains all review information. Let's not open up another, empty one.

derinb commented:
The problem I cant delete unnecessary commits on my remote master.

https://github.com/derinb/personas-plus/commits/master

I rebased and will try to run:
`git push -f origin master`

but it might overwrite the master and close the PR. Do you recommend to proceed?

wagnerand commented:
Ah you should have worked on a feature branch. What you need to do is note the commits you want to keep, then reset your master to this repository's master and then cherry-pick the commits you want to have.

derinb commented:
Thank you very much for the tip.

wagnerand commented on extension/bootstrap.js:
```diff
@@ -13,237 +13,237 @@ Components.utils.import("resource://gre/modules/XPCOMUtils.jsm");
+        domWindow.addEventListener("load", function listener() {
+            domWindow.removeEventListener("load", listener, false);
+            // If this is a browser window then setup its UI
+    
```
Nit: white-space only line. There might be more...

wagnerand commented on extension/bootstrap.js:
```diff
@@ -13,237 +13,237 @@ Components.utils.import("resource://gre/modules/XPCOMUtils.jsm");
+            catch (e) { }
+        }
+    },
+    shutdown:function(data,reason){
```
white-spaces missing. Again, use a formatter.

wagnerand commented on extension/bootstrap.js:
```diff
@@ -13,237 +13,237 @@ Components.utils.import("resource://gre/modules/XPCOMUtils.jsm");
+            Components.manager.addBootstrappedManifestLocation(data.installPath);
+    },
+    removeChromeProtocol:function(data) {
+        if (Services.vc.compare(Services.appinfo.platformVersion, "10.0") < 0 && Services.vc.compare(Services.appinfo.platformVersion, "8.0") >= 0)  
```
Let's split very long lines into two.

wagnerand commented on extension/bootstrap.js:
```diff
@@ -13,237 +13,237 @@ Components.utils.import("resource://gre/modules/XPCOMUtils.jsm");
+    removeDefaultLocalizations:function(){
+        this.STRINGS=[];
+        Components.classes["@mozilla.org/intl/stringbundle;1"].getService(Components.interfaces.nsIStringBundleService).flushBundles();
+    },  
```
trailing white-space.

wagnerand commented on extension/bootstrap.js:
```diff
@@ -13,237 +13,237 @@ Components.utils.import("resource://gre/modules/XPCOMUtils.jsm");
+    },
+    tearBrowserUI:function(){
+        var wm = Cc["@mozilla.org/appshell/window-mediator;1"].
+               getService(Ci.nsIWindowMediator);
```
Weird indentation.

wagnerand commented on extension/install.rdf:
```diff
@@ -7,7 +7,6 @@
     <em:version>1.7.7</em:version>
     <em:description>Your Firefox, your style. Easy-to-change themes.</em:description>
 	<em:bootstrap>true</em:bootstrap>
```
Wrong indentation.

wagnerand commented on extension/chrome.manifest:
```diff
@@ -82,3 +82,13 @@ locale   personas   uk-UA       locale/uk-UA/
+locale   personas   pt-PT       locale/pt-PT/
+locale   personas   sl-SI       locale/sl-SI/
+locale   personas   sr-RS       locale/sr-RS/
+locale   personas   vi-VN       locale/vi-VN/
```
Good catch, I told my self "Remember to update `chrome.manifest` when copying the locale files, but forgot anyway :(

derinb commented:
I already fixed instrall.rdf indentation.

https://github.com/mozilla/personas-plus/pull/24/commits/d75f9f5395659702a1e00d5c8ad790aa1e4c9f2c#diff-667ebbfc17994fceaf5ac79f769b62b9

Do not you have the changes on your side?

wagnerand commented:
I don't really like the mixing of real property getters and former properties you converted to methods in `PersonaController`. We should use either, but not both.

wagnerand commented:
> I already fixed instrall.rdf indentation.

Yeah I missed that because it was in another commit and I was looking at the changes per commit to make the review feasible.

wagnerand commented:
Most of the issues left are style/format issues, you should really use a formatter instead of trying to find and fix each issue manually.

wagnerand commented:
Please also remove the `!important` statements from the css files. You said yourself it works fine without them on Windows. And you didn't test on Mac, Linux. So until we have proof those are necessary, we shouldn't add them.

derinb commented:
> I don't really like the mixing of real property getters and former properties you converted to methods in PersonaController. We should use either, but not both.
Hello Andreas,

Any specific line please?

Getters probably placed for performance improvement. So I did not want to touch already working getters if there is no need. But other getters had to be modified because of PersonaController moved to bootstrap.js

wagnerand commented:
> Any specific line please?

You will easily find them by looking at the `PersonaController` definition.

derinb commented:
> You will easily find them by looking at the PersonaController definition.

Other getters are for common properties that are being used by the code. They do not depend on window or document. So for performance help, I left them as they are.

wagnerand commented on extension/bootstrap.js:
```diff
@@ -74,176 +76,192 @@ var PersonasPlusBootstrapAddon = {
+        var rh = ios.getProtocolHandler("resource").QueryInterface(Components.interfaces.nsIResProtocolHandler);
+        var nfu = ios.newFileURI(data.installPath);
+        var isDir = data.installPath.isDirectory();
+        if (isDir) { nfu = ios.newURI(nfu.spec + "/", null, null); } else { nfu = ios.newURI("jar:" + nfu.spec + "!/", null, null); }
```
proper condition format please

wagnerand commented on extension/modules/personas.js:
```diff
@@ -1178,7 +1178,7 @@ var PersonaController = {
         item.setAttribute("class", "menuitem-iconic");
         item.setAttribute("label", this._strings.get("viewMoreFrom", [category]));
-        item.addEventListener("command",function(event){PersonaController.openURLInTab(this.getAttribute('href'),event)},true);
+        item.addEventListener("command", function(event) { PersonaController.openURLInTab(this.getAttribute('href'), event) }, true);
```
consistent formatting please. You split similar lines like L1214.

derinb replied:
these "items" are created on the fly when popupshpwing and removed when onpopuphiding. So listeners only added once.

l1214 is for the nodes that are created by window load. when popupshowing, adding event listeners to already existent nodes by anonymous functions multiplies listeners. So I ported them to PersonaController.

I think on the fly items does not need this formatting. It will just increase PersonaController code and thus memory unnecessarily. 

What do you think?

wagnerand replied:
Code formatting should be consistent. Anonymous functions should have the same style everywhere.

wagnerand commented:
Looks good, only nits left!

wagnerand commented:
> Other getters are for common properties that are being used by the code. They do not depend on window or document. So for performance help, I left them as they are.

I'd still prefer consistent code over questionable performance differences, but we can leave it as it is, for now.

derinb commented:
Andreas, as expected, toolbar button's `-moz-box-orient` property is overwritten on Mac as `vertical`.
https://github.com/mozilla/personas-plus/blob/9c855c3edb0631e51d8246e93c651f2e6afa2ec4/extension/content/personas.css#L53

So button is shown as vertically on Mac rather than horizontal specified in CSS.

I think all CSS properties should have `!important;` to avoid any possible conflicts.

wagnerand commented:
> I think all CSS properties should have `!important;` to avoid any possible conflicts.

That does not fix the issue, rather it works-around it, which is not a good solution.

Using `AUTHOR_SHEET` instead of `USER_SHEET` when loading and unload the stylesheet actually fixes it.

wagnerand commented:
Ok, now please squash all commits into one.

wagnerand commented:
This looks good, thank you for your work!

derinb commented:
Thank you very much Andreas for your help.
